### PR TITLE
Allocation history

### PR DESCRIPTION
**Pull request by Mjwillis**
First pass on this new screen. Not linked to any other yet and subject to Eddie’s design changes

### CODE REVIEW

File: src/AllocationHistory/components/AllocationHistory.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+      currentKw = topEntry.expired ? '--' : properCaseName(topEntry.lastName) + ', ' + properCaseName(topEntry.firstName);
+    }
+    return currentKw;
+  }
+  render () {
+    const offenderDisplayName = properCaseName(this.props.allocationHistory.offender.firstName) + ' ' + properCaseName(this.props.allocationHistory.offender.lastName);
+    const allocations = this.props.allocationHistory.allocationHistory.map((a, index) => {
+      const kwName = properCaseName(a.lastName) + ', ' + properCaseName(a.firstName);
+      const createdStaff = properCaseName(a.createdByUser.lastName) + ', ' + properCaseName(a.createdByUser.firstName);
+      const lastModStaff = properCaseName(a.lastModifiedByUser.lastName) + ', ' + properCaseName(a.lastModifiedByUser.firstName);
```
Comment (steverendell) [inline src/AllocationHistory/components/AllocationHistory.js]: Could a.lastModifiedByUser be undefined?

File: src/AllocationHistory/components/AllocationHistory.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+            <tbody>{allocations}</tbody>
+          </table>
+        </div>
+      </div>
+    </div>
+    );
+
+    return (
+      <div>
+        <MessageBar {...this.props}/>
```
Comment (steverendell) [inline src/AllocationHistory/components/AllocationHistory.js]: dont think you need MessageBar , this is the green 'toast' update confirmation

File: src/AllocationHistory/containers/AllocationHistoryContainer.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  }
+
+  async getAllocationHistory () {
+    const response = await axios.get('/api/allocationHistory', {
+      params: {
+        offenderNo: this.props.match.params.offenderNo,
+      }
+    });
+
+    await this.props.allocationHistoryDispatch(response.data);
```
Comment (steverendell) [inline src/AllocationHistory/containers/AllocationHistoryContainer.js]: this 'await' is redundant

File: src/AllocationHistory/components/AllocationHistory.js (hunk 2 above)
Comment (Mjwillis) [reply src/AllocationHistory/components/AllocationHistory.js]: ok will sort out

File: src/AllocationHistory/components/AllocationHistory.js (hunk 1 above)
Comment (Mjwillis) [reply src/AllocationHistory/components/AllocationHistory.js]: It seems to be always set, but will add defensive code!

**Pull request merged**